Sources/TuistDependencies/Carthage/Models/CarthageVersionFile.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import TuistGraph
+
+/// A model that represents the `Carhtage` version file.
+/// Reference: https://github.com/Carthage/Carthage/blob/master/Documentation/VersionFile.md#version-files
+struct CarthageVersionFile: Decodable, Equatable {
+    enum CodingKeys: String, CodingKey {
+        case commitish
+        case iOS
+        case macOS = "Mac"
+        case watchOS
+        case tvOS
+    }
+
+    let commitish: String
+    let iOS: [Product]
+    let macOS: [Product]
+    let watchOS: [Product]
+    let tvOS: [Product]
+
+    /// Returns all products.
+    var allProducts: [Product] {
+        iOS + macOS + watchOS + tvOS
+    }
+}
+
+// MARK: - Models
+
+extension CarthageVersionFile {
+    struct Product: Decodable, Equatable {
+        let swiftToolchainVersion: String
+        let hash: String
+        let name: String
+        let container: String
+        let identifier: String
+
+        /// Returns architectures the product is built for.
+        var architectures: [BinaryArchitecture] {
```

Comment:
I was wondering what do we use this for? Can we make some validations based on this or?

Reply:
> Can we make some validations based on this or? 

Maybe... 🤔 

> I was wondering what do we use this for?

I assumed it will be mapped into something similar to `ValueGraphDependency.xcframework` when we start to marge graphs.

The `ValueGraphDependency.xcframework` contains the `XCFrameworkInfoPlist` and `XCFrameworkInfoPlist` contains [architectures](https://github.com/tuist/tuist/blob/ba28383dc89aab1d411002c0303fe794e24db767/Sources/TuistGraph/Models/XCFrameworkInfoPlist.swift#L30) property.  

So if we can collect this information from `Carthage` installation why do not store it. 🤷‍♂️

Anyway, if you have a feeling it will be not necessary or it could be added later then I can remove it.

Reply:
good point, we might need for the xcframework plist, let's keep it 👍